pkg/internal/infrastructure/terraform.go

```diff
@@ -87,6 +92,20 @@ func ComputeTerraformerChartValues(
 		if config.Networks.CloudNAT.MinPortsPerVM != nil {
 			minPortsPerVM = *config.Networks.CloudNAT.MinPortsPerVM
 		}
+		if config.Networks.CloudNAT.NatIPNames != nil {
+			natIPNames := []string{}
+			nIPs := []string{}
+			natIPOutputList := []string{}
+			for _, v := range config.Networks.CloudNAT.NatIPNames {
+				natIPNames = append(natIPNames, v.Name)
+				nIPs = append(nIPs, fmt.Sprintf("\"${data.google_compute_address.%s.self_link}\"", v.Name))
```

Comment:
Again, why are you doing such things here in the Go code? IMO **only the NAT IP names should be handed over to the TF Helm chart**. Then you can do such things in the Helm chart itself which has the benefit that all the logic is at one place. WDYT?

Reply:
https://github.com/gardener/gardener-extension-provider-gcp/pull/39#discussion_r384894731

Reply:
Because otherwise i have to add a string join function and add it to the template function map, not sure if this is the better way.

Reply:
I don't get it, sorry. These Terraform expressions belong in the Terraform chart itself. Otherwise, if you would have to change  something in the future, you would have to do it in two places which is hard to maintain and error-prone.